components/sponsored_benefits/app/models/effective/datatables/broker_agency_plan_design_organization_datatable.rb

```diff
@@ -160,6 +160,17 @@ def check_employer_osse_eligibility(employer)
         osse_eligibility = bs&.current_eligibility(:osse_subsidy) if bs.present?
         osse_eligibility.present?
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        return false if current_user.blank?
+        return true if current_user.has_hbx_staff_role?
+
+        (current_user.has_broker_agency_staff_role? &&  ba_staff_role_belongs_to_profile?(current_user, attributes["profile_id"])) ? true : false
+      end
+
+      def ba_staff_role_belongs_to_profile?(current_user, ga_profile_id)
+        current_user.person.broker_agency_staff_roles.any? { |role| role.benefit_sponsors_broker_agency_profile_id == ga_profile_id }
+      end
```

Comment:
Don't we need to check if the roles are active?
And doesn't the ID come in as a string, where the `role.benefit_sponsors_broker_agency_profile_id` would be a BSON id?  So don't we need to convert to compare?
Shouldn't we be using a 'policies' model?  I notice there are no 'policies' present in this project.

Reply:
Don't we need to check if the roles are active?
- has_broker_agency_staff_role? method already checks for active staff role.

And doesn't the ID come in as a string, where the role.benefit_sponsors_broker_agency_profile_id would be a BSON id? So don't we need to convert to compare?
- When I validated through UI it did not throw error or failed authorization. But let me update the line to convert.

Shouldn't we be using a 'policies' model? I notice there are no 'policies' present in this project.
- There are no policies defined in sponsored_benefit components, I am not sure if adding policies model is a big lift.